### PR TITLE
Exposes getExperienceReward and patches an experience dupe in ZombieEntity

### PR DESCRIPTION
**Pull request by baileyholl**
This PR adds a new method to LivingEntity that calls the protected getExperienceRewards method and fires the forge getExperienceDrop event to return the 'true' exp value of a mob. The forge event in dropExperience has been replaced with this new method call that provides the same functionality. If shouldDropExperience returns false, this method returns 0.

One additional patch is made to ZombieEntity to remove the mutation of xpReward for babies, otherwise the experience value of the mob would increase each time getExperienceDrop is called. This change does not modify the experience value in any way in the super call, MobEntity#getExperienceReward where xpReward is simply checked for 0 and assigned to another int.

**Why a new method?**

Currently, there is no way to get the experience value of a mob until the forge event fires, when it has already died. getExpierenceDrop, shouldDropExperience, and xpReward are all set to protected in LivingEntity and MobEntity. Additionally, the xpReward  field is unreliable for getting the experience value of the mob, as it is not used at all for AnimalEntity and WaterMobEntity.

Changing getExperienceDrop and shouldDropExperience to public is a breaking change for 1.16, and should perhaps be considered for 1.17. Furthermore, even if these two methods were public, this method also combines the experience reward and the forge event for the modder without requiring them to manually fire the forge event each time.

### CODE REVIEW

**Review by marchermans — CHANGES_REQUESTED** [review]
This is backwards breaking in it's current state and won't be merged into 1.16 as it states right now. 

Since an RB is out you need to maintain backwards compatibility and that means that the old behaviour is still needed. You just need to mark it as depreciated.

**Comment by baileyholl** [comment]
I have reverted the patch to dropExperience in order to maintain the old behavior. What should be marked as depreciated? The other methods are inaccessible without AT/Mixin at the moment.

File: patches/minecraft/net/minecraft/entity/LivingEntity.java.patch
Comment (KnightMiner) [inline patches/minecraft/net/minecraft/entity/LivingEntity.java.patch]: This patch change is a bit redundant now, its just removing the `this`

**Review by marchermans — APPROVED** [review]
Ignore my stuff....
I misread the patch file.... twice.....

File: patches/minecraft/net/minecraft/entity/LivingEntity.java.patch
Comment (baileyholl) [reply patches/minecraft/net/minecraft/entity/LivingEntity.java.patch]: I have readded 'this' to match the previous patch.

**Comment by TheCurle** [comment]
I fail to see how "Changing getExperienceDrop and shouldDropExperience to public is a breaking change for 1.16". 
Things that relied on them being private would have no change, as a wider visibility is a superset rather than a subset.

Mods compiled against the new build would error on older versions of Forge, yes, but the same is true for every new feature we add in the release cycle. It's the reason we bump the RB every so often. 

Honestly, this could all just be one AT line and it'd be fine. But i'll defer to the actual maintainers of the project for final opinion.

**Comment by baileyholl** [comment]
Are existing binaries affected by changing this method from protected to public? Any mod that overrides that method will be doing so with a stricter access than the parent class. Is this a compile time check only? 

Additionally, while I originally planned this PR with only changing the access in mind, I think this method providing the event call for the modder and respecting canDropExperience is a nice quality of life addition.

**Comment by sciwhiz12** [comment]
Hello! Since the application of the Stale label, there has been no recent activity on this PR. As said in our previous comment, this PR will now be closed for inactivity.

If you wish to continue this PR, please open a new PR against the Latest supported version (to be conformant with our version support and contributions policy). 

Thank you for your contributions. 👋

**Pull request closed without merging**
